src/journal-remote/journal-remote-main.c

```diff
@@ -175,7 +180,25 @@ static int request_meta(void **connection_cls, int fd, char *hostname, Compressi
 
         log_debug("Added RemoteSource as connection metadata %p", source);
 
-        source->compression = compression;
+        for (int i = 0; arg_compression != NULL && arg_compression[i] != NULL; i++)
+                count++;
+
+        if (count > 0) {
+                float step = q / count;
+
+                for (int i = 0; arg_compression[i] != NULL; i++, q-=step) {
+                        if (buf == NULL)
+                                r = asprintf(&buf, "%s;q=%.1f", compression_lowercase_to_string(arg_compression[i]->algorithm), q);
+                        else
+                                r = asprintf(&buf, ",%s;q=%.1f", compression_lowercase_to_string(arg_compression[i]->algorithm), q);
+                        if (r < 0)
+                                return log_oom();
+                        source->encoding = strnappend(source->encoding, buf, r);
+                        free(buf);
+                }
+        }
```

Comment:
please split this out to a helper call, i.e. compile_content_encoding_string() or so, that generates this string